javatests/arcs/android/storage/service/StorageServiceEndToEndTest.kt

```diff
@@ -92,13 +104,61 @@ class StorageServiceEndToEndTest {
         DriverFactory.clearRegistrations()
     }
 
+    @Test
+    fun writeThenRead_nullInlines_inCollection_onDatabase() = runBlocking {
+        val handle = createCollectionHandle(databaseKey)
+        val entity = emptyEntityForTest()
+
+        handle.dispatchStore(entity)
+
+        val handle2 = createCollectionHandle(databaseKey)
+        val data = handle2.fetchAll()
+        assertThat(data.size).isEqualTo(1)
+        assertThat(data.toList()[0]).isEqualTo(entity)
+    }
+
     @Test
     fun writeThenRead_inlineData_inCollection_onDatabase() = runBlocking {
         val handle = createCollectionHandle(databaseKey)
         val entity = entityForTest()
 
         handle.dispatchStore(entity)
 
+        databaseManager.runGarbageCollection()
```

Comment:
can you run this line twice?
GarbageCollection deletes stuff across two runs (mark as orphan on first, actually delete on second)
(and in the test below as well)